### PR TITLE
WS-2 Adding new location

### PR DESCRIPTION
**Pull request by klezner**
Task "WS-2 Adding new location" from Epic "Weather management" finished and able to merge.

### CODE REVIEW

**Comment by MichalKulygin** [comment]
8 of 10 tests failed

File: src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    LocationService locationService;
+
+    @Test
+    void createLocation_thenCreatesNewLocation() {
+        // given
+        when(locationRepository.save(any())).thenReturn(LocationTestHelper.provideLocation());
+        // when
+        Location location = locationService.createLocation("Gdansk", "Pomeranian", "Poland", 54.3, 18.6);
+        //then
+        assertThat(location.getCity()).isEqualTo("Gdansk");
```
Comment (MisQl) [inline src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java]: you check the values created by you for testing purposes, it does not test the functionality of that class

next level, lets try to use `ArgumentCaptor` - https://www.baeldung.com/mockito-argumentcaptor

eg.
```
// when
locationService.createLocation("Gdansk", "Pomeranian", "Poland", 54.3, 18.6);
// then
verify(locationRepository).save(locationArgumentCaptor.capture());
assertThat(locationArgumentCaptor.getValue().getRegion()).isEqualTo("Pomeranian");

File: src/main/java/pl/kl/weatherservice/location/CreateLocationRequest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import javax.validation.constraints.*;
+
+@Getter
+@Builder
+class CreateLocationRequest {
+
+    @NotNull(message = "City is mandatory!")
+    @NotEmpty(message = "City is mandatory!")
+    @NotBlank(message = "City is mandatory!")
```
Comment (MichalKulygin) [inline src/main/java/pl/kl/weatherservice/location/CreateLocationRequest.java]: NotBlank does the job

**Review by MichalKulygin — APPROVED** [review]

File: src/main/java/pl/kl/weatherservice/location/CreateLocationRequest.java (hunk 2 above)
Comment (klezner) [reply src/main/java/pl/kl/weatherservice/location/CreateLocationRequest.java]: Good point. Fixed.

File: src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java (hunk 1 above)
Comment (klezner) [reply src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java]: ArgumentCaptor implemented.

File: src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Test
+    void createLocation_thenCreatesNewLocation() {
+        // given
+        when(locationRepository.save(any())).thenReturn(LocationTestHelper.provideLocation());
+        // when
+        locationService.createLocation("Gdansk", "Pomeranian", "Poland", 54.3, 18.6);
+        //then
+        verify(locationRepository).save(locationArgumentCaptor.capture());
+        assertThat(locationArgumentCaptor.getValue().getCity().equals("Gdansk"));
```
Comment (MisQl) [inline src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java]: create a variable for `locationArgumentCaptor.getValue()`

**Review by MisQl — APPROVED** [review]

File: src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java (hunk 3 above)
Comment (klezner) [reply src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java]: Fixed.

**Pull request merged**